library/src/test/java/com/novoda/downloadmanager/DownloadManagerTest.java

```diff
@@ -0,0 +1,281 @@
+    private final FileOperations fileOperations = mock(FileOperations.class);
+    private final DownloadsBatchPersistence downloadsBatchPersistence = mock(DownloadsBatchPersistence.class);
+    private final LiteDownloadManagerDownloader downloadManagerDownloader = mock(LiteDownloadManagerDownloader.class);
+
+    private DownloadManager downloadManager;
+    private Map<DownloadBatchId, DownloadBatch> downloadBatches = new HashMap<>();
+    private List<DownloadBatchStatus> downloadBatchStatuses = new ArrayList<>();
+    private List<DownloadBatchCallback> downloadBatchCallbacks = new ArrayList<>();
+
+    @Before
+    public void setUp() {
+        downloadBatches = new HashMap<>();
+        downloadBatches.put(DOWNLOAD_BATCH_ID, downloadBatch);
+        downloadBatches.put(ADDITIONAL_DOWNLOAD_BATCH_ID, additionalDownloadBatch);
+
+        downloadBatchCallbacks.add(downloadBatchCallback);
+
+        downloadManager = new DownloadManager(
+                lock,
+                executorService,
+                downloadBatches,
+                downloadBatchCallbacks,
+                fileOperations,
+                downloadsBatchPersistence,
+                downloadManagerDownloader
+        );
+
+        setupDownloadBatchesResponse();
+        setupDownloadBatchStatusesResponse();
+
+        given(downloadBatch.status()).willReturn(BATCH_STATUS);
+        given(additionalDownloadBatch.status()).willReturn(ADDITIONAL_BATCH_STATUS);
+    }
+
+    private void setupDownloadBatchesResponse() {
+        final ArgumentCaptor<DownloadsBatchPersistence.LoadBatchesCallback> loadBatchesCallbackCaptor = ArgumentCaptor.forClass(DownloadsBatchPersistence.LoadBatchesCallback.class);
+        willAnswer(new Answer<Void>() {
+            @Override
+            public Void answer(InvocationOnMock invocation) throws Throwable {
+                loadBatchesCallbackCaptor.getValue().onLoaded(Arrays.asList(downloadBatch, additionalDownloadBatch));
+                return null;
+            }
+        }).given(downloadsBatchPersistence).loadAsync(any(FileOperations.class), loadBatchesCallbackCaptor.capture());
+    }
+
+    private void setupDownloadBatchStatusesResponse() {
+        final ArgumentCaptor<List> argumentCaptor = ArgumentCaptor.forClass(List.class);
+        willAnswer(new Answer<Void>() {
+            @Override
+            public Void answer(InvocationOnMock invocation) throws Throwable {
+                downloadBatchStatuses = argumentCaptor.getValue();
+                return null;
+            }
+        }).given(allBatchStatusesCallback).onReceived(argumentCaptor.capture());
+    }
+
+    @Test
+    public void setDownloadService_whenInitialising() {
+        downloadManager.initialise(downloadService);
+
+        verify(downloadManagerDownloader).setDownloadService(downloadService);
+    }
+
+    @Test(timeout = 500)
+    public void notifyAll_whenInitialising() throws InterruptedException {
```

Comment:
Can use `timeout` because `synchronized` with `wait` and `notify` will block and unblock the testing thread. 

We force the testing thread to lock-up with `lock.wait` and at the same time schedule a call that will internally release the thread with `downloadManager.initialise`.